### ISSUE TITLE
Updating Repeater Sub Field Value based on Unique ID

### ISSUE THREAD

**Issue by prodjallen** [AdvancedCustomFields/acf#153]
Is it possible to update a sub field in a repeater field set by using the unique_id that belongs to the repeater field set, not the row count typically used in "update_sub_field"?

Since the order of the array can change, based on the sorting in the admin area, i'd like to be able to do this if its possible, say increment a repeater sub field, having the unique_id?

![image](https://user-images.githubusercontent.com/48658997/54515527-896ad200-4919-11e9-9139-f34f21cd4a69.png)

Like in the image / example above, to use the Unique_ID of the sub field to update the value of the up count vote to say increment by 1 using ajax or something along those lines.

Since the sort order would change the row count and render using the sort / row count in the array useless. Might ask how? The row count becomes useless in a situation like this since a viewer may load a page, admin changes sort order, then when ajax is fired at a later time when front-end elements are already loaded, it would end up updating the wrong field.

In PHP and mysql this is very trivial, but it being in the wordpress environment, need to be cautious. If you have any insight on how this maybe possible i'd greatly appreciate it.

**Comment by elliotcondon** [comment]
Hi @prodjallen 

Thanks for the question.

To answer your quesiton, not, it is not possible to update a repeater sub field value without knowing the row number. 

If you have 3 rows of data, which row do you want to update? 1, 2 or 3?

That said, you can find the "row" dynamically by searching for your "unique_id".
By loading the repeater field value, you can loop over the existing values and find the row that matches your "unique_id" value. This will give you the row number, and you can then update the value.

If performance is something you are concerned about, don't load the entire repeater field value.
Instead, use SQL to search the postmeta table for the entry where the meta_value == "unique_id_value". This will give you access to the meta_key which will look something like `repeater_2_unique_id`.

From this "key", you can load the other values like `repeater_2_up_vote_count`, and then modify their values.

In short, it is possible, but will require lots of custom PHP and SQL. Good luck!

**Comment by prodjallen** [comment]
Thanks for your help!

I was finally able to make a more optimized version of doing this, i'm sure a lot of people have this question as well...

Here it is:
$allfields = get_field('repeater');
$uniqueids = array_column($allfields, 'unique_id');
			$found_row_id = array_search('your_unique_id_to_row', $uniqueids);
			if($found_row_id != null){
				$the_row = $found_row_id + 1;
				$newvalue = '14';
				update_sub_field( array('repeater', $the_row, 'up_vote_count'), $newvalue);
			}

repeater: The repeater name
unique_id: A random unique value generated with each row via addon by https://github.com/KLicheR/wp-acf-unique_id.
your_unique_id_to_row: The unique id from above to locate the row number.
up_vote_count: This was just my sub field name in the repeater that needed updating.
$newvalue is fairly trivial but its the new value of the row - sub field being updated.

A similar function built into ACF for users would be nice to have!

-Allen
